cmd/bundle/generate/job.go

```diff
@@ -50,10 +50,15 @@ func NewGenerateJobCommand() *cobra.Command {
 		}
 
 		downloader := newDownloader(w, sourceDir, configDir)
-		for _, task := range job.Settings.Tasks {
-			err := downloader.MarkTaskForDownload(ctx, &task)
-			if err != nil {
-				return err
+		// Don't download files if the job is using Git source
+		if job.Settings.GitSource != nil {
+			cmdio.LogString(ctx, "Job is using Git source, skipping downloading files")
+		} else {
+			for _, task := range job.Settings.Tasks {
+				err := downloader.MarkTaskForDownload(ctx, &task)
```

Comment:
What if the notebook_task has `source = WORKSPACE`?

Reply:
You mean if it's a mix of git sourced job and tasks with workspace files? Not sure if it's possible, but even if it is I'd say we better keep it "all or nothing" approach

Reply:
Yeah, it's possible. According to the `git_source` API docs:
```
If git_source is set, these tasks retrieve the file from the remote repository by default. However, this behavior can be overridden by setting source to WORKSPACE on the task.
```

I'd presume mixed approaches would be common, but IDK. Do we have reason to believe they are uncommon?

Reply:
It should be simple to handle either way, so it seems more of a design choice here.

Reply:
From UI/Workspace perspective it's indeed a bit more straightforward because workspace source notebooks have absolute workspace path and git ones - relative paths. But it's more complex when we generate bundle YAML config and download files: we make all paths relatives and such a mix can be a bit more tricky to handle correctly. Since we recommend not to use Git jobs generally, I'd suggest to keep it simple and not download anything

Reply:
I agree we shouldn't support it, but if the job contains a mix of `source` attributes, then the generated job won't be deployable. Would it be possible to enforce that `source` is empty or equal to "GIT" for all tasks?

Reply:
> Would it be possible to enforce that source is empty or equal to "GIT" for all tasks?

Good point, will update to handle this